driver-core/src/main/java/com/datastax/driver/core/BoundStatement.java

```diff
@@ -222,9 +222,12 @@ public BoundStatement bind(Object... values) {
      * precedence. If the routing key has been set through {@link PreparedStatement#setRoutingKey} then that is used
      * next. If neither of those are set then it is computed.
      *
+     * @param protocolVersion unused by this implementation (no internal serialization is required to compute the key).
+     * @param codecRegistry unused by this implementation (no internal serialization is required to compute the key).
+     *
      * @return the routing key for this statement or {@code null}.
      */
-    public ByteBuffer getRoutingKey() {
+    public ByteBuffer getRoutingKey(ProtocolVersion protocolVersion, CodecRegistry codecRegistry) {
```

Comment:
Missing `@Override`. We also could propose an overloaded version with no arguments.